### PR TITLE
Feature/selection strat roulette

### LINKED ISSUE #7: Selection Strategy
Implement a slection strategy e.g. Roulette wheel selection.

### PR DESCRIPTION
**Pull request by Persists**
### Summary
This PR introduces the `rouletteWheelSelection` function, which selects two parents from a population based on their fitness scores using the roulette wheel selection algorithm.

### CODE REVIEW

**Comment by Persists** [comment]
I'm not entirely sure if all these exceptions are needed.

File: src/Selection.jl [hunk 1]
```diff
@@ -2,6 +2,54 @@ module Selection
 
 using ..Types
 
-# TODO
+function rouletteWheelSelection(population::Vector{Vector{Int64}}, fitness_scores::Vector{Float64}, rand_generator::Function = rand)::Tuple{Vector{Int64}, Vector{Int64}}
```
Comment (Daniel1402) [inline src/Selection.jl]: We can select more than Int64 Chormosomes. I would change it to Matrix{T} and also the return type to Tuple{Vector{T}, Vector{T}} where T <: Number.

Comment (Daniel1402) [inline src/Selection.jl]: Function Documentation missing.

File: src/Selection.jl [hunk 2]
```diff
@@ -2,6 +2,54 @@ module Selection
...
+
+
+        @info "Fitness Scores: $fitness_scores"
+        @info "Cumulative Probabilities: $cum_probs"
+            
+        fixed_rand = rand_generator()
+
+        selected_index = findfirst(cum_prob -> fixed_rand <= cum_prob, cum_probs)
+
+        return selected_index == nothing ? length(fitness_scores) : selected_index
```
Comment (Daniel1402) [inline src/Selection.jl]: Use `isnothing` to compare against nothing.

File: src/GeneticAlgorithms.jl [hunk 3]
```diff
@@ -79,5 +79,5 @@ function optimize(
 end
 
 
-export optimize, Crossover
+export optimize, Crossover, Selection, timesTwo
```
Comment (Daniel1402) [inline src/GeneticAlgorithms.jl]: `timesTwo`?

File: src/Selection.jl (hunk 1 above)
Comment (Daniel1402) [inline src/Selection.jl]: Why do we need a rand_generator? Just for testing?

Comment (Daniel1402) [reply src/Selection.jl]: -> Just looked up how to test random stuff. Your way seems fine.

**Review by Daniel1402 — CHANGES_REQUESTED** [review]

File: src/GeneticAlgorithms.jl (hunk 3 above)
Comment (Persists) [reply src/GeneticAlgorithms.jl]: fail

File: src/Selection.jl (hunk 1 above)
Comment (Persists) [reply src/Selection.jl]: yes its for testing, i didn,t find a propper way to inject a custom generator

File: src/Selection.jl (hunk 2 above)
Comment (Persists) [reply src/Selection.jl]: thx, nice one

File: src/Selection.jl (hunk 1 above)
Comment (Persists) [reply src/Selection.jl]: please review if the new docs are enough

**Review by Daniel1402 — APPROVED** [review]

**Pull request merged**